jsonapi/jsonapi_utility_blackbox_test.go

```diff
@@ -0,0 +1,70 @@
+package jsonapi_test
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"testing"
+
+	"github.com/almighty/almighty-core/app"
+	"github.com/almighty/almighty-core/errors"
+	"github.com/almighty/almighty-core/jsonapi"
+	"github.com/almighty/almighty-core/resource"
+	"github.com/stretchr/testify/require"
+)
+
+func TestErrorToJSONAPIError(t *testing.T) {
+	t.Parallel()
+	resource.Require(t, resource.UnitTest)
+
+	var jerr app.JSONAPIError
+	var httpStatus int
+
+	// test not found error
+	jerr, httpStatus = jsonapi.ErrorToJSONAPIError(errors.NewNotFoundError("foo", "bar"))
+	require.Equal(t, http.StatusNotFound, httpStatus)
+	require.NotNil(t, jerr.Code)
+	require.NotNil(t, jerr.Status)
+	require.Equal(t, jsonapi.ErrorCodeNotFound, *jerr.Code)
+	require.Equal(t, strconv.Itoa(httpStatus), *jerr.Status)
+
+	// test not found error
+	jerr, httpStatus = jsonapi.ErrorToJSONAPIError(errors.NewConversionError("foo"))
+	require.Equal(t, http.StatusBadRequest, httpStatus)
+	require.NotNil(t, jerr.Code)
+	require.NotNil(t, jerr.Status)
+	require.Equal(t, jsonapi.ErrorCodeConversionError, *jerr.Code)
+	require.Equal(t, strconv.Itoa(httpStatus), *jerr.Status)
+
+	// test bad parameter error
+	jerr, httpStatus = jsonapi.ErrorToJSONAPIError(errors.NewBadParameterError("foo", "bar"))
+	require.Equal(t, http.StatusBadRequest, httpStatus)
+	require.NotNil(t, jerr.Code)
+	require.NotNil(t, jerr.Status)
+	require.Equal(t, jsonapi.ErrorCodeBadParameter, *jerr.Code)
+	require.Equal(t, strconv.Itoa(httpStatus), *jerr.Status)
+
+	// test internal server error
+	jerr, httpStatus = jsonapi.ErrorToJSONAPIError(errors.NewInternalError("foo"))
+	require.Equal(t, http.StatusInternalServerError, httpStatus)
+	require.NotNil(t, jerr.Code)
+	require.NotNil(t, jerr.Status)
+	require.Equal(t, jsonapi.ErrorCodeInternalError, *jerr.Code)
+	require.Equal(t, strconv.Itoa(httpStatus), *jerr.Status)
```

Comment:
I check that the returned `httpStatus` is for instance `http.StatusNotFound`. Then I check that the converted error as a string matches the error code that is stored in the `app.JSONAPIError`. I think that implicitly checks for the correct error code, don't you think?